### PR TITLE
Add staking transaction struct and handler logic

### PR DESCRIPTION
**Pull request by rlan35**
Tested locally with cross-shard txn. It still works.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -255,6 +260,17 @@ func (node *Node) reducePendingTransactions() {
 	}
 }
 
+func (node *Node) reducePendingStakingTransactions() {
+	txPoolLimit := core.ShardingSchedule.MaxTxPoolSizeLimit()
+	curLen := len(node.pendingStakingTransactions)
+
+	// If length of pendingStakingTransactions is greater than TxPoolLimit then by greedy take the TxPoolLimit recent transactions.
+	if curLen > txPoolLimit+txPoolLimit {
+		node.pendingStakingTransactions = append(types2.StakingTransactions(nil), node.pendingStakingTransactions[curLen-txPoolLimit:]...)
```
Comment (chaosma) [inline node/node.go]: we can first detect duplicated staking transactions, then remove the older transactions

Comment (chaosma) [reply node/node.go]: e.g.  remove the duplicated staking in add pending transactions function to save memory

File: node/worker/worker.go [hunk 2]
```diff
@@ -96,13 +98,6 @@ func (w *Worker) throttleTxs(selected types.Transactions, recentTxsStats types.R
 
 // SelectTransactionsForNewBlock selects transactions for new block.
 func (w *Worker) SelectTransactionsForNewBlock(newBlockNum uint64, txs types.Transactions, recentTxsStats types.RecentTxsStats, txsThrottleConfig *shardingconfig.TxsThrottleConfig, coinbase common.Address) (types.Transactions, types.Transactions, types.Transactions) {
-	// Must update to the correct current state before processing potential txns
-	if err := w.UpdateCurrent(coinbase); err != nil {
```
Comment (chaosma) [inline node/worker/worker.go]: why removed

File: staking/types/staking_transaction.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"math/big"
+
+	"github.com/harmony-one/harmony/internal/common"
+)
+
+// StakingTransaction struct.
+type StakingTransaction struct {
+	AccountNonce uint64         `json:"nonce"      gencodec:"required"`
```
Comment (chaosma) [inline staking/types/staking_transaction.go]: renamed to Nonce ?

Comment (rlan35) [reply staking/types/staking_transaction.go]: Just following the transaction field. and This is the same as tx's nonce.

File: node/worker/worker.go (hunk 2 above)
Comment (rlan35) [reply node/worker/worker.go]: Moved it out of this func. so it applied to both normal tx and staking tx.

File: node/node.go (hunk 1 above)
Comment (rlan35) [reply node/node.go]: Done.

**Pull request merged**